darts/config.py

```diff
@@ -0,0 +1,448 @@
+            raise_log(
+                ValueError(
+                    "option_context requires an even number of arguments (option-value pairs)"
+                ),
+                logger,
+            )
+
+        # Store original values
+        original_values = {}
+        try:
+            # Set new values
+            for pattern, value in zip(args[::2], args[1::2]):
+                original_values[pattern] = self.get_option(pattern)
+                self.set_option(pattern, value)
+
+            yield
+        finally:
+            # Restore original values
+            for key, value in original_values.items():
+                self.set_option(key, value)
+
+
+# Global options manager instance
+_global_options = _OptionsManager()
+
+
+# Public API functions
+def get_option(pat: str) -> Any:
+    """
+    Retrieves the value of the specified option.
+
+    Available Options:
+
+    - display.[max_rows, max_cols]
+    - plotting.use_darts_style
+
+    Parameters
+    ----------
+    pat
+        The option key to retrieve. Must uniquely identify a single option.
+
+    Returns
+    -------
+    Any
+        The current value of the option.
+
+    Raises
+    ------
+    ValueError
+        If no option matches the pattern, or if the pattern is ambiguous.
+
+    Examples
+    --------
+    >>> from darts import get_option
+    >>> get_option('display.max_rows')
+    10
+    """
+    return _global_options.get_option(pat)
+
+
+def set_option(pat: str, value: Any) -> None:
+    """
+    Sets the value of the specified option.
+
+    Available Options:
+
+    - display.[max_rows, max_cols]
+    - plotting.use_darts_style
+
+    Parameters
+    ----------
+    pat
+        The option key to set. Must uniquely identify a single option.
+    value
+        The new value for the option. Must be valid according to the option's validator.
+
+    Raises
+    ------
+    ValueError
+        If no option matches the pattern, if the pattern is ambiguous, or if the value is invalid for the option.
+
+    Examples
+    --------
+    >>> from darts import set_option
+    >>> set_option('display.max_rows', 20)
+    """
+    _global_options.set_option(pat, value)
+
+
+def reset_option(pat: str) -> None:
+    """
+    Reset one or more options to their default value.
+
+    Available Options:
+
+    - display.[max_rows, max_cols]
+    - plotting.use_darts_style
+
+    Parameters
+    ----------
+    pat
+        The option key or pattern to reset. Can match multiple options. Use `"all"` to reset all options.
+
+    Raises
+    ------
+    ValueError
+        If no option matches the pattern.
+
+    Examples
+    --------
+    >>> from darts import reset_option
+    >>> reset_option('display.max_rows')  # Reset single option
+    >>> reset_option('display')  # Reset all display options
+    >>> reset_option('all')  # Reset all options
+    """
+    _global_options.reset_option(pat)
+
+
+def describe_option(pat: str) -> Optional[str]:
```

Comment:
method always returns a string, shouldn't be Optional
```suggestion
def describe_option(pat: str) -> str:
```